File: DS/ASSIGNMENT2/DSASSIGNMENT2_SECOND/src/Queue.java

public interface Queue {
	/**
	 * This method add element to the queue if its not full
	 * @param value integer value to be added to queue
	 * @return true if value is added else false
	 * @throws Exception when queue is full
	 */
	public boolean enQueue(int value)throws Exception;
	
	/**
	 * This method remove element from the queue
	 * @return element which is removed
	 * @throws Exception when queue is empty
	 */
	public int deQueue()throws Exception;
	
	/**
	 * This method check if the queue is empty
	 * @return true if queue is empty else false
	 */
	public boolean isEmpty();
	
	/**
	 * This method check if the queue is full
	 * @return true if queue is full else false
	 */
	public boolean isFull();
}

File: UI/ASSIGNMENT8/queue.js
function queue() {
    this.front = 0;
    this.rear = 0;
    this.data = {};
}

//function to enqueue an element
queue.prototype.enqueue = function (element) {
    this.data[++this.rear] = element;
    if (this.front == 0) {
        this.front++;
    }
}

//function to dequeue from queue
queue.prototype.dequeue = function () {
    if (this.front == 0 || (this.front > this.rear)) {
        return "queue is empty";
    }
    var element = this.data[this.front];
    delete this.data[this.front];
    this.front++;
    return element;
}